### ISSUE TITLE
Whitespaces in config cause errors

### ISSUE THREAD

**Issue by roelofr** [emersion/mako#282]
I've been struggling with this issue, until I looked into the `config.c` file, but it seems that trailing whitespace breaks some config values (like colors).

## Expectations

No error is thrown with this config

```conf
background-color=#FF0000AA 
```

(there's a space after the hex)

## Result

> `Failed to parse option 'background-color'`

Likely has to do with the string length check here:
https://github.com/emersion/mako/blob/3aba143ff3f3c2acdabf86a3b6755dc55bf7248d/types.c#L45-L48

Maybe the values read from the config should be trimmed?

**Comment by roelofr** [comment]
Of course, removing trailing whitespace is good practice as a user, but it shouldn't cause errors if it didn't happen.

**Comment by emersion** [comment]
See https://github.com/emersion/mako/issues/277

**Comment by roelofr** [comment]
I did see that issue, but figured that'd be about ` = ` vs `=`. Different issue, same result.

**Comment by TheAvidDev** [comment]
The resolution depends on what we want to apply this to. Currently, `format` and (maybe) `icon-path` can have trailing whitespace. At least for `format`, I don't think there's really any effect on rendering. I'm not sure of the effect on `icon-path`. If we are okay with stripping whitespace on *both*, then this is probably an alright suggestion considering it could be applied to all options. 

However, if we allow whitespace on `format` and `icon-path` but strip it from all other options, the code would become more complicated and I'm not sure if this would be a great idea.

**Comment by th0rn** [comment]
A simple solution would be to just change the error to:

Failed to parse option 'background-color' value '#FF0000AA '

That way it would be obvious to also those that don't have trailing space highlighting in their editors.

**Comment by TheAvidDev** [comment]
> I'm not sure of the effect on `icon-path`. 

Just thinking back on this, issues would only arise if your icon path had a space on the end, in which case.... why does your file name contain a space with nothing after it?

Therefore, applying this stripping to all options seems alright.

**Comment by emersion** [comment]
Fixed in https://github.com/emersion/mako/pull/411
